src/test/java/org/candlepin/subscriptions/resource/InstancesResourceTest.java

```diff
@@ -0,0 +1,106 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.when;
+
+import java.time.OffsetDateTime;
+import java.util.*;
+import org.candlepin.subscriptions.db.AccountListSource;
+import org.candlepin.subscriptions.db.HostRepository;
+import org.candlepin.subscriptions.db.model.Host;
+import org.candlepin.subscriptions.resteasy.PageLinkCreator;
+import org.candlepin.subscriptions.security.WithMockRedHatPrincipal;
+import org.candlepin.subscriptions.tally.AccountListSourceException;
+import org.candlepin.subscriptions.utilization.api.model.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.test.context.ActiveProfiles;
+
+@SpringBootTest
+@ActiveProfiles({"api", "test"})
+@WithMockRedHatPrincipal("123456")
+class InstancesResourceTest {
+
+  @MockBean HostRepository repository;
+  @MockBean PageLinkCreator pageLinkCreator;
+  @MockBean AccountListSource accountListSource;
+  @Autowired InstancesResource resource;
+
+  @BeforeEach
+  public void setup() throws AccountListSourceException {
+    when(accountListSource.containsReportingAccount("account123456")).thenReturn(true);
+  }
+
+  @Test
+  void testShouldPopulateInstanceResponse() {
+    var host = new Host();
+    host.setInstanceId("d6214a0b-b344-4778-831c-d53dcacb2da3");
+    host.setDisplayName("rhv.example.com");
+    host.setLastSeen(OffsetDateTime.now());
+
+    Mockito.when(
+            repository.findAllBy(
+                any(), any(), any(), any(), any(), anyInt(), anyInt(), any(), any(), any()))
+        .thenReturn(new PageImpl<>(List.of(host)));
+    var data = new InstanceData();
+    data.setId(host.getInstanceId());
+    data.setDisplayName(host.getDisplayName());
+    data.setLastSeen(host.getLastSeen());
+
+    var meta = new InstanceMeta();
+    meta.setCount(1);
+    meta.setProduct(ProductId.RHOSAK);
+    meta.setServiceLevel(ServiceLevelType.PREMIUM);
+    meta.setUsage(UsageType.PRODUCTION);
+    meta.setMeasurements(List.of("Instance-hours", "Storage-gibibytes", "Transfer-gibibytes"));
+
+    var expected = new InstanceResponse();
+    expected.setData(List.of(data));
+    expected.setMeta(meta);
+
+    InstanceResponse report =
+        resource.getInstancesByProduct(
+            RHOSAK,
+            null,
+            null,
+            ServiceLevelType.PREMIUM,
+            UsageType.PRODUCTION,
+            null,
+            null,
+            null,
+            InstanceReportSort.DISPLAY_NAME,
+            null);
+
+    // Reassigned values for measurements as logic cannot be mocked correctly.
```

Comment:
Can you elaborate on what can't be mocked correctly?  It feels weird to take the "actual" value, and then overwrite it before comparing it to "expected".

Reply:
The `asTallyHostViewApiInstance` method is used to populate measurements, If I remove the line that creates a empty list the method would return a list of nulls `[null,null,null]`for each of the uoms. 
I can't populate the expected with a list of nulls. So I opted to use a work around.

Reply:
I'm going to look into this a little more locally.  If we want/expect an empty list rather than a list of nulls (which makes sense to me), my gut is telling me to look into the logic returning the list of nulls and maybe update it to return an empty list instead of hacking the unit tests.

Thoughts? @ksynvrit @kahowell

Reply:
> I can't populate the expected with a list of nulls.

Why not? I would expect nulls here actually.

Reply:
I think if a measurement isn't there it should be null. Like in the example output in [ENT-4383](https://issues.redhat.com/browse/ENT-4383). The way it's implemented now. I believe it's consistent with the example in the card. I don't want to adjust the implementation of the logic to return an empty list, if it's only relevant for the unit test. If there's a case where we would want to return a empty list, than I' all for add the logic to return an empty list.

Reply:
> > I can't populate the expected with a list of nulls.
> 
> Why not? I would expect nulls here actually.
I'm working on adjusting it so excepted get a list of nulls. It's was just I could directly add a list of nulls from the setter method.

Reply:
I agree with not changing the implementation to return an empty list, based on the additional information.

Since this is the case, please do not override the returned data before asserting equals.  Update the expected `data` to include a list of three null values.  I added a simple example of how to achieve this using github's suggestion feature at line 77.